common/src/main/java/com/wynntils/models/character/CharacterModel.java

```diff
@@ -71,7 +74,8 @@ public final class CharacterModel extends Model {
     private boolean reskinned;
     private int level;
 
-    private boolean isVeteran = false;
+    @Persisted
+    private final Storage<Boolean> isVeteran = new Storage<>(false);
```

Comment:
Also this wouldnt get checked if silverbull is still active, so I made it persistent

Reply:
I agree with this, good catch.